Userland/Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -1270,42 +1271,44 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Event>> Document::create_event(DeprecatedSt
 
     // 2. If interface is an ASCII case-insensitive match for any of the strings in the first column in the following table,
     //      then set constructor to the interface in the second column on the same row as the matching string:
-    auto interface_lowercase = interface.to_lowercase();
-    if (interface_lowercase == "beforeunloadevent") {
+    if (Infra::is_ascii_case_insensitive_match(interface, "beforeunloadevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create BeforeUnloadEvent
-    } else if (interface_lowercase == "compositionevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "compositionevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create CompositionEvent
-    } else if (interface_lowercase == "customevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "customevent"sv)) {
         event = CustomEvent::create(realm, "");
-    } else if (interface_lowercase == "devicemotionevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "devicemotionevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DeviceMotionEvent
-    } else if (interface_lowercase == "deviceorientationevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "deviceorientationevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DeviceOrientationEvent
-    } else if (interface_lowercase == "dragevent") {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "dragevent"sv)) {
         event = Event::create(realm, ""); // FIXME: Create DragEvent
-    } else if (interface_lowercase.is_one_of("event", "events")) {
+    } else if (Infra::is_ascii_case_insensitive_match(interface, "evesvnt"sv)
```

Comment:
:mag_right: "evesvnt"

Reply:
Let's add a template helper? For cases like this where we compare to multiple things:

```c++
template<typename... Ts>
bool is_ascii_case_insensitive_match_to_one_of(StringView string, Ts&&... strings) 
{
    return (... || is_ascii_case_insensitive_match(string, forward<Ts>(strings)));
}
```

Reply:
Thank you!!! My multi-cursor editing is a bit funky sometimes.